src/coreclr/debug/dbgutil/machoreader.cpp

```diff
@@ -126,19 +126,36 @@ MachOModule::TryLookupSymbol(const char* symbolName, uint64_t* symbolValue)
         _ASSERTE(m_nlists != nullptr);
         _ASSERTE(m_strtabAddress != 0);
 
-        for (int i = 0; i < m_dysymtabCommand->nextdefsym; i++)
+        // First, search just the "external" export symbols 
+        if (TryLookupSymbol(m_dysymtabCommand->iextdefsym, m_dysymtabCommand->nextdefsym, symbolName, symbolValue))
         {
-            std::string name = GetSymbolName(i);
-            // Skip the leading underscores to match Linux externs
-            if (name[0] == '_')
-            {
-                name.erase(0, 1);
-            }
-            if (strcmp(name.c_str(), symbolName) == 0)
-            {
-                *symbolValue = m_loadBias + m_nlists[i].n_value;
-                return true;
-            }
+            return true;
+        }
+        // If not found in external symbols, search all of them
+        if (TryLookupSymbol(0, m_symtabCommand->nsyms, symbolName, symbolValue))
+        {
+            return true;
+        }
+    }
+    *symbolValue = 0;
+    return false;
+}
+
+bool
+MachOModule::TryLookupSymbol(int start, int nsyms, const char* symbolName, uint64_t* symbolValue)
+{
+    for (int i = 0; i < nsyms; i++)
+    {
+        std::string name = GetSymbolName(start + i);
+        // Skip the leading underscores to match Linux externs
+        if (name[0] == '_')
+        {
+            name.erase(0, 1);
```

Comment:
I know this is there from before:

`const char* currentName = name.length() > 0 && name[0] == '_' ? name.c_str() + 1 : name.c_str();` 

GetSymbolName is also a little allocation heavy, but that's a future thing.

Reply:
Might be better to avoid the extra allocation if we are risking iterating over all the symbols and not just the exports.